### PR TITLE
Issue 340 fix output tests

### LINKED ISSUE #340: Fix output tests for DFN, Composite and Reaction-Diffusion
- DFN: needs #333 to fix particles 
- Composite: not sure what the problem is
- Reaction-diffusion: shouldn't be a `LeadAcidBaseModel`

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Fix output tests for all models except `NewmanTiedemannCapacitance`.
Still a few things to fix for lead-acid, but opening a PR as there is already a fair bit in here
Fixes #340 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks: 

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pybamm/models/base_models.py [hunk 1]
```diff
@@ -183,6 +186,15 @@ def jacobian(self):
     def jacobian(self, jacobian):
         self._jacobian = jacobian
 
+    @property
+    def use_capacitance(self):
+        return self._use_capacitance
+
+    @use_capacitance.setter
+    def use_capacitance(self, use_capacitance):
```
Comment (martinjrobins) [inline pybamm/models/base_models.py]: where is `use_capacitance` used? I'm guessing that it governs the equations that are put in the model (during __init__). So would calls to this setter after the model is created have any effect? Should this be a read-only option?

File: tests/integration/test_models/standard_output_comparison.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        self.compare("Average open circuit voltage")
+        # Currents
+        self.compare("Average negative electrode interfacial current density")
+        self.compare("Average positive electrode interfacial current density")
+        # Concentration
+        self.compare("Average electrolyte concentration")
+
+
+class VariablesComparison(BaseOutputComparison):
+    "Compare variables that vary across models"
```
Comment (martinjrobins) [inline tests/integration/test_models/standard_output_comparison.py]: I don't get this.... doesn't `self.compare` test that the variables are similar across models? I feel I'm missing something obvious

File: pybamm/models/base_models.py (hunk 1 above)
Comment (tinosulzer) [reply pybamm/models/base_models.py]: Yes, will change this to read-only

File: tests/integration/test_models/standard_output_comparison.py (hunk 2 above)
Comment (tinosulzer) [reply tests/integration/test_models/standard_output_comparison.py]: Yeah re-reading now this is bad wording ... the distinction between `VariablesComparison` and `AveragesComparison` is that averages should be the same across models regardless of current, whereas variables will differ a bit and only be approximately equal for certain (low) values of the current. 
May this should just be "compare variables across models"?

Comment (tinosulzer) [reply tests/integration/test_models/standard_output_comparison.py]: Yeah re-reading now this is bad wording ... the distinction between VariablesComparison and AveragesComparison is that averages should be the same across models regardless of current, whereas variables will differ a bit and only be approximately equal for certain (low) values of the current.
May this should just be "compare variables across models"?

Comment (tinosulzer) [reply tests/integration/test_models/standard_output_comparison.py]: @martinjrobins - I have changed to "compare variables across models", let me know if this is ok and I will merge

**Comment by tinosulzer** [comment]
Cool, will merge this. The problem was reading `phi_e` before updating it with the composite solution, i.e. using `phi_e` from the leading-order model to calculate the voltage

**Pull request merged**